instance/models/openedx_instance.py

```diff
@@ -392,3 +396,63 @@ def delete(self, *args, **kwargs):
         self.deprovision_s3()
         self.deprovision_rabbitmq()
         super().delete(*args, **kwargs)
+
+    def update_consul_metadata(self):
+        """
+        This method is going over all fields in this model to reflect
+        their values on Consul.
+        Values on consul will be fetched first, then updated only if
+        they changed in this version of the model using agent's `cas`
+        parameter (Check-And-Set). If PUT operation successfully
```

Comment:
It appears this isn't accurate anymore.  Is there any reason you abandoned that approach?

Reply:
Outdated documentation, yes. I didn't (some how) abandoned the approach, but instead, I handled it in Python instead of Consul. The idea here now instead of getting the keys one-by-one which hits the consul multiple times, I fetched them all at once, and did Check-And-Set in a python for-loop that compares the checked value with the currently calculated one.

Reply:
This is also gonna be helpful in removing unnecessary fields that might go stale.